lib/charms/istio_k8s/v0/istio_ingress_config.py

```diff
@@ -155,32 +151,29 @@ def publish(
             databag.update(data)
             log.debug("Published provider data: %s to relation: %s", data, relation)
 
-    def get_ext_authz_provider_name(self) -> Optional[str]:
+    def get_ext_authz_provider_name(self, relation: Relation) -> Optional[str]:
```

Comment:
This relation handler already knows the `relation` object (it knows `charm_relation_mapping` and `relation_name`), so no need to provide `relation` here